lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -578,6 +587,95 @@ def partial_update(self, request, comment_id):
         return Response(update_comment(request, comment_id, request.data))
 
 
+class CanPostOrComment(permissions.BasePermission):
+    """
+    Permission that checks to see if the user is allowed to post or
+    comment in the course.
+    """
+
+    def has_permission(self, request, view):
+        """Returns true if the user is enrolled or is staff."""
+        course_key = CourseKey.from_string(view.kwargs.get('course_id'))
+        return (
+            GlobalStaff().has_user(request.user) or
+            CourseStaffRole(course_key).has_user(request.user) or
+            CourseInstructorRole(course_key).has_user(request.user) or
+            CourseEnrollment.is_enrolled(request.user, course_key) or
+            has_discussion_privileges(request.user, course_key)
+        )
+
+
+class UploadFileView(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Upload a file to be attached to a thread or comment.
+
+    **URL Parameters**
+
+        * course_id:
+            The ID of the course where this thread or comment belongs.
+
+    **POST Upload File Parameters**
+
+        * thread_key:
+            If the upload belongs to a comment, refer to the parent
+            `thread_id`, otherwise it should be `"root"`.
+
+    **Example Requests**:
+        POST /api/discussion/v1/courses/{course_id}/upload/
+        Content-Type: multipart/form-data; boundary=--Boundary
+
+        ----Boundary
+        Content-Disposition: form-data; name="thread_key"
+
+        <thread_key>
+        ----Boundary
+        Content-Disposition: form-data; name="uploaded_file"; filename="<filename>.<ext>"
+        Content-Type: <mimetype>
+
+        <file_content>
+        ----Boundary--
+
+    **Response Values**
+
+        * location: The URL to access the uploaded file.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthentication,
+        SessionAuthentication,
+    )
+    permission_classes = (
+        permissions.IsAuthenticated,
+        CanPostOrComment,
+    )
+
+    def post(self, request, course_id):
+        """
+        Handles a file upload.
+        """
+        self.check_permissions(request)
```

Comment:
Why you are calling that explicitly? It should be Django Rest Frameworks' job to call that internally. Do you think its not working as expected and you have to call it explicitly?

Reply:
You're right, this is not required.